src/support/mod.rs

```diff
@@ -137,10 +138,62 @@ pub fn get_llvm_version() -> (u32, u32, u32) {
     return (major, minor, patch);
 }
 
-pub fn load_library_permanently(filename: &str) -> bool {
-    let filename = to_c_str(filename);
+/// Possible errors that can occur when loading a library
+#[derive(thiserror::Error, Debug, PartialEq, Eq, Clone, Copy)]
+pub enum LoadLibraryError {
+    /// The given path could not be converted to a [`&str`]
+    #[error("The given path could not be converted to a `&str`")]
+    UnicodeError,
+    /// The given path could not be loaded as a library
+    #[error("The given path could not be loaded as a library")]
+    LoadingError,
+}
+
+/// Permanently load the dynamic library at the given `path`.
+///
+/// It is safe to call this function multiple times for the same library.
+pub fn load_library_permanently(path: &Path) -> Result<(), LoadLibraryError> {
+    let filename = to_c_str(path.to_str().ok_or(LoadLibraryError::UnicodeError)?);
+
+    let error = unsafe { LLVMLoadLibraryPermanently(filename.as_ptr()) == 1 };
+    if error {
+        return Err(LoadLibraryError::LoadingError);
+    }
+
+    Ok(())
+}
+
+#[test]
+fn test_load_library_permanently() {
+    assert_eq!(
+        load_library_permanently(Path::new("missing.dll")),
+        Err(LoadLibraryError::LoadingError)
+    );
+}
+
+/// Permanently loads all the symbols visible inside the current program
+pub fn load_visible_symbols() {
+    unsafe { LLVMLoadLibraryPermanently(std::ptr::null()) };
+}
```

Comment:
I didn't find it anywhere in docs, but people from StackOverflow (and tests) say that passing `nullptr` as filename results in loading all the symbols visible inside program